src/daemon/DaemonApp.cpp

```diff
@@ -119,6 +120,12 @@ namespace SDDM {
     int DaemonApp::newSessionId() {
         return m_lastSessionId++;
     }
+
+    void DaemonApp::keyboardLayout(const QString &layout) {
+        OrgFreedesktopLocale1Interface locale1(QStringLiteral("org.freedesktop.locale1"), QStringLiteral("/org/freedesktop/locale1"),
+                                                   QDBusConnection::systemBus());
+        locale1.SetX11Keyboard(layout, {}, {}, {}, false, false);
```

Comment:
Is there a reason the greeter can't make the call directly without proxying through the server?

Reply:
>Interactive authentication required.

It needs root, fair enough.